packages/kinvey-datastore/src/repositories/network-repository.js

```diff
@@ -0,0 +1,99 @@
+import { KinveyRequest, RequestMethod, DeltaFetchRequest } from 'kinvey-request';
+
+import { Repository } from './repository';
+import { buildCollectionUrl, ensureArray } from '../utils';
+
+/**
+ * @typedef RequestOptions
+ * @property {Operation} operation
+ * @property {Object} options
+ * @property {Query} [options.query]
+ * @property {Query} [options.query]
+ * @property {Object} [options.aggregation]
+ * @property {Number} [options.timeout]
+ * @property {Object} [options.properties]
+ * @property {Boolean} [options.skipBL]
+ * @property {Boolean} [options.trace]
+ */
+
+export class NetworkRepository extends Repository {
+  read(collection, query, options) {
+    const requestConfig = this._buildRequestConfig(collection, RequestMethod.GET, null, query, null, options = {});
+    return this._makeHttpRequest(requestConfig, options.useDeltaFetch);
+  }
+
+  readById(collection, entityId, options) {
+    const requestConfig = this._buildRequestConfig(collection, RequestMethod.GET, null, null, entityId, options);
+    return this._makeHttpRequest(requestConfig);
+  }
+
+  create(collection, entities, options) {
+    return this._processBatch(collection, RequestMethod.POST, entities, null, options);
+  }
+
+  update(collection, entities, options) {
+    return this._processBatch(collection, RequestMethod.PUT, entities, options);
+  }
+
+  deleteById(collection, entityId, options) {
+    const requestConfig = this._buildRequestConfig(collection, RequestMethod.DELETE, null, null, entityId, options);
+    return this._makeHttpRequest(requestConfig);
+  }
+
+  delete(collection, query, options) {
+    const requestConfig = this._buildRequestConfig(collection, RequestMethod.DELETE, null, query, null, options);
+    return this._makeHttpRequest(requestConfig);
+  }
+
+  count(collection, query, options) {
+    const requestConfig = this._buildRequestConfig(collection, RequestMethod.GET, null, query, '_count', options);
+    return this._makeHttpRequest(requestConfig);
+  }
+
+  _processBatch(collection, method, entities, options) {
+    const isSingle = !Array.isArray(entities);
+    const requestPromises = ensureArray(entities).map((entity) => {
+      const id = method === RequestMethod.PUT ? entity._id : null; // TODO: this isn't great :)
+      const requestConfig = this._buildRequestConfig(collection, method, entity, null, id, options);
+      return this._makeHttpRequest(requestConfig);
+    });
+    // TODO: different error handling?
+    return Promise.all(requestPromises)
+      .then(res => (isSingle ? res && res[0] : res));
+  }
+
+  _makeHttpRequest(requestConfig, deltaFetch) {
+    if (deltaFetch) { // TODO: maybe this needs to be in sync manager
```

Comment:
Needs to be thought out. This is too deep in the stack for the delta set logic. I'd expect it to be implemented in the "processor", since it utilizes both the cache and the network. Also, `_makeHttpRequest` isn't the right place anyway, since even `delete` seems to use it.

Reply:
This is temporary - it's how I reuse deltaset fetching. I can refactor it a bit, but we can reuse all the logic for deltaset and it is out of scope for this, so I think this is the best way to go about it. This `deltaFetch` is the flag I was talking about in the first comment, about it being the only option being passed along.

As for the eventual place for it, I thought it's in the `SyncManager` class, as I believed it was only relevant for pull. But it turns out it's also relevant for `CacheStore.find()`. It used to be in `NetworkStore.find()` too, but I think it shouldn't be there. So the minimum change required at this point is to add support for deltaset fetching for `CacheStore.find()`. I think the rest can be left for when we start optimizing for performance, as deltaset needs to be reworked in general. Up to you, really.

Reply:
Agree with the second part. Delta set logic shouldn't be in the `NetworkStore`. 

As for your comment about when we refactor it to its "eventual" solution - please open a JIRA ticket with your comment as the context. These are the types of gaps in our current implementation that I'm hoping to uncover during the redesign.

Reply:
I meant more the functionality, than the logic code. I mean NetworkStore should not be able to do a deltaset find. And I think we all agree on that.

Where the logic goes would become apparent when we decide on who should be able to use it. And I think only `pull()` should use it.

Should `CacheStore.find()` be able to use deltaset fetching? What should the response of that be - should it only contain the changed entities, or all entities matching the query?

`CacheStore`'s responses are still a bit of a mystery to me. I view it as a `NetworkStore`, which first gives you the results currently in the cache, and then updates the cache with whatever it got from network. Is that a fair assumption? Is this the way it should throw errors - only based on network operations? I guess in this sense, it shouldn't be able to do a deltaset `find`. Perhaps this should be an internal optimization on how it does a `find`, but shouldn't be a public API/setting?